### ISSUE TITLE
WebGL mask+filter issue

### ISSUE THREAD

**Issue by darkwhite** [GoodBoyDigital/pixi.js#843]
Can not use any filter in WebGL if a mask is applied to sprite or any container in hierarchy.

The following example shows the issue:

    var stage0 = new PIXI.Stage(0x66FF99);

	var rendererType=1;
	var addMask=true;
	
	var renderer =  null;
	if(rendererType==0)
	{
		renderer =  new PIXI.CanvasRenderer(800, 600, null, false);
	}
	else
	{
		renderer =  PIXI.autoDetectRenderer(800, 600, null, false);
	}

	document.body.appendChild(renderer.view);
	
	var container0= new PIXI.DisplayObjectContainer();
	stage0.addChild(container0);
	
	if(addMask)
	{
		var clippingMask0 = new PIXI.Graphics();
		clippingMask0.beginFill();
		clippingMask0.drawRect(0, 0, 600, 300);
		clippingMask0.endFill();
		container0.mask = clippingMask0;
		container0.addChild(clippingMask0);
	}

	var texture0 = PIXI.Texture.fromImage("http://www.goodboydigital.com/pixijs/examples/14/panda.png");

	var sprite0 = new PIXI.Sprite(texture0);
	container0.addChild(sprite0);
	var fltr = new PIXI.GrayFilter();
	sprite0.filters = [fltr];

	requestAnimFrame(animate);	

	function animate() {
	    requestAnimFrame(animate);
	    renderer.render(stage0);
	}

As can observe, the sprite is not visible! If use Canvas renderer by setting "rendererType=0;" or use WebGl renderer but don't add the mask (by setting "addMask=false;"), the sprite will be visible.

**Comment by GoodBoyDigital** [comment]
This code runs like a boss in v3. :)

**Comment by darkwhite** [comment]
Yes, it is fixed! 
Thank you : )

**Comment by Dinkelborg** [comment]
Could it be that this issue is back in Version 4? 
I'm new to Pixi, but I've been searching for a week at least trying to find any hints on why I can only apply filters to my graphic object directly. 

I create a circle texture in code like this: 
`var graphic = new PIXI.Graphics();
    graphic.lineStyle(0);
    graphic.boundsPadding = 20;
    graphic.beginFill(liquidColor, 1);
    graphic.drawCircle(0, 0, 3);
    graphic.endFill();
    graphic.filters = [ blurFilter ];

    this.texture = this.app.renderer.generateTexture(graphic);`

This works fine and the blur is also applied, but no matter where, if I try to apply the blur to the container, the stage, any extra container I create - it all either does absolutely nothing, or nothing is visible at all any more but the background color of the stage. 
> No console errors btw.

**Comment by ivanpopelyshev** [comment]
That's not supposed to happen. Make a fiddle and I'll look whats wrong. Which version is it?

**Comment by bigtimebuddy** [comment]
FYI @Dinkelborg, keep in mind [filters won't show up on your root stage object](https://github.com/pixijs/pixi.js/wiki/v4-Gotchas#filters-on-root-pixicontainer-dont-render).

**Comment by Dinkelborg** [comment]
Hey @ivanpopelyshev, 
I made a narrowed down version of my project here: [https://codepen.io/Dinkelborg/pen/dVBaWd](https://codepen.io/Dinkelborg/pen/dVBaWd)
I marked the lines with comments where I tried to apply the filter (lines: 12,16,19,28,45)
I didn't expect each of those to work and as @bigtimebuddy mentioned, I also found the Gotcha page for v4 so I expected that filters on stage wouldn't work, I was just trying what I could ... basically what I did expect to work was the "spriteContainer" (line 16) 
Basically as I understood the info I found online I should be able to apply any filter on any container as long as it's not the stage root one or a particle container, right?

**Comment by ivanpopelyshev** [comment]
May be this will help you: particleContainer has no way to know what bounds are, because it will affect performance. Thus, you should manually set filterArea on container, you can take whole screen if you want.

```js
container.filters = [...];
container.filterArea = renderer.screen; //same as app.screen, its Rectangle
```

**Comment by Dinkelborg** [comment]
@ivanpopelyshev 
Awesome - this fixed it! 
Yeah it would be really cool if this could be added to the filter docs maybe even highlighted as a tip - so that if you are looking up how a filter works it will remind you that you have to set a filter area for your container.

Thanks a lot for the quick help
